include/swift/SIL/SILInstruction.h

```diff
@@ -6378,16 +6377,34 @@ class UncheckedOwnershipConversionInst
 /// the base must not be moved before any instructions which depend on the
 /// result of this instruction, exactly as if the address had been obviously
 /// derived from that operand (e.g. using ``ref_element_addr``). The result is
-/// always equal to the first operand.
+/// always equal to the first operand and thus forwards ownership through the
+/// first operand. This is a "regular" use of the second operand (i.e. the
```

Comment:
This statement contradicts the comments above: "A single value inst that forwards a static ownership from one (or all) of its operands."

Also, I would avoid calling the second argument a regular use. It is regular only in that it does not forward or consume its operand. But it is not a point-in-time use. It is essentially a use of indeterminate scope that does not require ownership. (This will make a lot more sense when we have begin/end_dependence and we can just call it a scoped use).

Anyway, what we have here is:

OwnershipForwardingSingleValue forwards ownership of one of its operands (is it always the first?, how to we identify it?) to its single result.

The first MarkDependence operand's ownership is forwarded to the result.

The second MarkDependence operand is neither consumed, nor forwarded. It guarantees liveness up to the transitive use points.

Reply:
> This statement contradicts the comments above: "A single value inst that forwards a static ownership from one (or all) of its operands."

See my comment above. I meant it in a more general sense that it is derived from merging the ownership of a subset of the arguments.

> 
> Also, I would avoid calling the second argument a regular use. It is regular only in that it does not forward or consume its operand. But it is not a point-in-time use. It is essentially a use of indeterminate scope that does not require ownership. (This will make a lot more sense when we have begin/end_dependence and we can just call it a scoped use).

I was trying to say that in the ownership model it /is/ a point use. That being said, I agree the right way to talk about it is as a scoped use (which would eliminate the inconsistency). So do you have any thoughts on what to say in the mean time.

> 
> Anyway, what we have here is:
> 
> OwnershipForwardingSingleValue forwards ownership of one of its operands (is it always the first?, how to we identify it?) to its single result.

It is defined as a special case for MarkDependence since it isn't merge all of the operands. The special case here is partly why it was missed, so your right about the code smell. Do you have a better idea on how to do this? My naive first instinct was to try to create a generic way for a SILInstruction to express what parameters that it forwards from. That would allow this to generalize to MultipleValueInstructionResults. Today they have a similar problem except they combine a ValueOwnershipKind with a Type.

> 
> The first MarkDependence operand's ownership is forwarded to the result.
> 
> The second MarkDependence operand is neither consumed, nor forwarded. It guarantees liveness up to the transitive use points.